src/package.lisp

```diff
@@ -28,6 +28,14 @@
       package-designator
       (oget *package-table* (string package-designator))))
 
+(defun delete-package (package-designator)
+  ;; TODO: Signal a correctlable error in case the package-designator does not
+  ;; name a package.
+  ;; TODO: Implement unuse-package and remove the deleted package from packages
+  ;; that use it.
+  (delete-property (package-name (find-package package-designator))
```

Comment:
This should be `find-package-or-fail`, right?

Also, string should fail if its argument is not a string designator. Now in JSCL `(find-package 3)` evaluates to NIL instead of signaling an error.